client-js/app/elements/labrad-grapher.ts

```diff
@@ -95,35 +130,51 @@ export class LabradGrapher extends polymer.Base {
             tags: ['^trash']
           });
         }
+        this.set('listItems.' + selectedIndex + '.trashed',
+                 !this.selected.trashed);
+
+        // Whenever we toggle trash, it will be removed from the view.
+        if (this.filterListItems_([this.selected]).length == 0) {
+          this.splice('listItems', selectedIndex, 1);
+        }
         break;
     }
   }
 
-  @computed()
-  listItems(
-    path: Array<string>,
-    dirs: Array<{name: string; url: string; tags: Array<string>}>,
-    datasets: Array<{name: string; url: string; tags: Array<string>}>,
-    kick: number
-  ) {
-    if (!dirs) dirs = [];
-    if (!datasets) datasets = [];
-
-    var isTagged = (item: {tags: Array<string>}, tag: string) => {
-      for (let t of item.tags) {
-        if (t === tag) return true;
-      }
-      return false;
-    }
-    var isIncluded = (item: {tags: Array<string>}) => {
-      if (this.showStars) return isTagged(item, 'star');
-      if (!this.showTrash) return !isTagged(item, 'trash');
-      return true;
-    }
+  newDir(dir: {name: string; url: string; tags: Array<string>}): void {
+    const obj = {
+      id: 'dir:' + dir.name,
+      name: dir.name,
+      url: dir.url,
+      isDir: true,
+      isDataset: false,
+      isParent: false,
+      starred: this.isTagged_(dir, 'star'),
+      trashed: this.isTagged_(dir, 'trash')
+    };
+    this.addItemToList_(obj);
```

Comment:
Originally obj was being passed to more than one function, a relic I forgot to reconsider. I agree it would be nicer to just pass the literal directly.